geomesa-core/src/test/scala/org/locationtech/geomesa/core/iterators/MapAggregatingIteratorTest.scala

```diff
@@ -0,0 +1,344 @@
+    )
+
+  val TEST_TABLE_NAME = "map_aggregating_iterator_test"
+
+  def createDataStore(i: Int): DataStore = {
+    val mockInstance = new MockInstance("dummy" + i)
+    val c = mockInstance.getConnector("user", new PasswordToken("pass".getBytes))
+    val tableName = s"$TEST_TABLE_NAME$i"
+    c.tableOperations.create(tableName)
+    val splits = (0 to 99).map(s => "%02d".format(s)).map(new Text(_))
+    c.tableOperations().addSplits(tableName, new java.util.TreeSet[Text](splits.asJava))
+
+    val dsf = new AccumuloDataStoreFactory
+
+    import org.locationtech.geomesa.core.data.AccumuloDataStoreFactory.params._
+
+    val ds = dsf.createDataStore(Map[String, JSerializable](
+      zookeepersParam.key -> "dummy",
+      instanceIdParam.key -> "dummy",
+      userParam.key       -> "user",
+      passwordParam.key   -> "pass",
+      tableNameParam.key  -> tableName,
+      mockParam.key       -> "true").asJava)
+    ds
+  }
+
+  def loadFeatures(ds: DataStore, sft: SimpleFeatureType, encodedFeatures: Array[_<:Array[_]]): SimpleFeatureStore = {
+    val builder = AvroSimpleFeatureFactory.featureBuilder(sft)
```

Comment:
Just a warning, the kryo encoding doesn't play nice with non-sequential unit tests, as the mock connection will return the exact same byte array object for a given key/value pair, which messes up kryo as it tries to manipulate the byte array.